src/Runner.Common/JobServer.cs

```diff
@@ -117,6 +135,19 @@ private async Task CheckNetworkEndpointsAsync(int attemptsLeft)
             }
         }
 
+        public void InitializeWebsocketClient(ServiceEndpoint serviceEndpoint)
+        {
+             this._serviceEndpoint = serviceEndpoint;
+             InitializeWebsocketClient(TimeSpan.Zero);
+        }
+
+        public ValueTask DisposeAsync()
+        {
+            Trace.Info($"Disposing websocket client ...");
+            this._websocketClient?.CloseOutputAsync(WebSocketCloseStatus.NormalClosure, "Shutdown", CancellationToken.None);
+            return ValueTask.CompletedTask;
```

Comment:
do you need to call `GC.SuppressFinalize(this);`?

Reply:
I remember we call it in other files of the runner.

Reply:
done! Kind of surprised though, I never thoughts we will be poking with `GC`, but looks like the pattern here, so added